### PR TITLE
Small improvements to the devcontainer

### PR DESCRIPTION
**Pull request by Nereuxofficial**
This removes a workaround no longer necessary to install LLVM and updates the rust-analyzer plugin to use the official name, although the old one is aliased to my knowledge. 

A possible improvement to the build time of the Devcontainer image would be to utilize [cargo-binstall](https://github.com/cargo-bins/cargo-binstall) to download the binary release for sccache. Let me know if that is a worthwhile improvement and i'll add it.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -22,12 +22,11 @@ RUN rustup component add rustfmt clippy
...
 RUN apt update && apt install -y build-essential gdb git wget python3-venv ninja-build lsb-release software-properties-common gnupg cmake
-# Workaround until https://github.com/llvm/llvm-project/issues/62475 is resolved
 RUN set -ex &&\
-    echo "deb http://apt.llvm.org/bookworm/ llvm-toolchain-bookworm-${LLVM_VERSION} main" > /etc/apt/sources.list.d/apt.llvm.org.list &&\
-    wget -qO- https://apt.llvm.org/llvm-snapshot.gpg.key |  tee /etc/apt/trusted.gpg.d/apt.llvm.org.asc &&\
-    apt update &&\
-    apt-get install -y clang-${LLVM_VERSION} lldb-${LLVM_VERSION} lld-${LLVM_VERSION} clangd-${LLVM_VERSION} clang-tidy-${LLVM_VERSION} clang-format-${LLVM_VERSION} clang-tools-${LLVM_VERSION} llvm-${LLVM_VERSION}-dev lld-${LLVM_VERSION} lldb-${LLVM_VERSION} llvm-${LLVM_VERSION}-tools libomp-${LLVM_VERSION}-dev libc++-${LLVM_VERSION}-dev libc++abi-${LLVM_VERSION}-dev libclang-common-${LLVM_VERSION}-dev libclang-${LLVM_VERSION}-dev libclang-cpp${LLVM_VERSION}-dev libunwind-${LLVM_VERSION}-dev libclang-rt-${LLVM_VERSION}-dev libpolly-${LLVM_VERSION}-dev
+    wget https://apt.llvm.org/llvm.sh &&\
+    chmod +x llvm.sh &&\
+    ./llvm.sh
```
Comment (tokatoka) [inline Dockerfile:28]: don't you have to select the version here?
iirc it'll download the latest (19) if not specified
but yes using llvm.sh is always much better.

another point is that this won't link /usr/bin/clang to /usr/bin/clang-{xx} automatically. unlike installing by `apt-get install`.
if you want the symlink to be correctly made you can use something this 
https://gist.github.com/tokatoka/26f4ba95991c6e33139999976332aa8e
(but you have to decide version)

Comment (tokatoka) [reply Dockerfile:28]: no download is not good.
i think you should just copy paste or the shell script or write it out and execute it inside the dockerfile.

File: scripts/createAliases.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+# creates a symbolic link from bin-x.x to bin
+# This just strips off last 3 characters when creating a link
+
+LLVMFILES=/usr/bin/llvm*
+CLANGFILES=/usr/bin/clang*
+LLC=/usr/bin/llc-17
+OPT=/usr/bin/opt-17
+LLD=/usr/bin/lld-17
```
Comment (tokatoka) [inline scripts/createAliases.sh:10]: this 17 has to be given by the argument

**Comment by domenukk** [comment]
Shellcheck isn't happy yet about the `createAliases` script:

```sh
shellcheck ./scripts/*.sh

In ./scripts/createAliases.sh line 6:
LLVMFILES=/usr/bin/llvm*
          ^------------^ SC2[12](https://github.com/AFLplusplus/LibAFL/actions/runs/10873386066/job/30169439681?pr=2522#step:7:13)5 (warning): Brace expansions and globs are literal in assignments. Quote it or use an array.


In ./scripts/createAliases.sh line 7:
CLANGFILES=/usr/bin/clang*
           ^-------------^ SC2125 (warning): Brace expansions and globs are literal in assignments. Quote it or use an array.


In ./scripts/createAliases.sh line [15](https://github.com/AFLplusplus/LibAFL/actions/runs/10873386066/job/30169439681?pr=2522#step:7:16):
	echo "linking" $f "to" $link
                       ^-- SC2086 (info): Double quote to prevent globbing and word splitting.
                               ^---^ SC2086 (info): Double quote to prevent globbing and word splitting.

Did you mean: 
	echo "linking" "$f" "to" "$link"


In ./scripts/createAliases.sh line [16](https://github.com/AFLplusplus/LibAFL/actions/runs/10873386066/job/30169439681?pr=2522#step:7:17):
	ln -s $f $link
              ^-- SC2086 (info): Double quote to prevent globbing and word splitting.
                 ^---^ SC2086 (info): Double quote to prevent globbing and word splitting.

Did you mean: 
	ln -s "$f" "$link"


In ./scripts/createAliases.sh line [17](https://github.com/AFLplusplus/LibAFL/actions/runs/10873386066/job/30169439681?pr=2522#step:7:18):
	if [ -e $f ]
                ^-- SC2086 (info): Double quote to prevent globbing and word splitting.

Did you mean: 
	if [ -e "$f" ]


In ./scripts/createAliases.sh line [18](https://github.com/AFLplusplus/LibAFL/actions/runs/10873386066/job/30169439681?pr=2522#step:7:19):
	  then cp $link /usr/local/bin/
                  ^---^ SC2086 (info): Double quote to prevent globbing and word splitting.

Did you mean: 
	  then cp "$link" /usr/local/bin/

For more information:
  https://www.shellcheck.net/wiki/SC2125 -- Brace expansions and globs are li...
  https://www.shellcheck.net/wiki/SC[20](https://github.com/AFLplusplus/LibAFL/actions/runs/10873386066/job/30169439681?pr=2522#step:7:21)86 -- Double quote to prevent globbing ...
Error: Process completed with exit code 1.
```

**Comment by Nereuxofficial** [comment]
> Shellcheck isn't happy yet about the `createAliases` script:
> 
> ```shell
> shellcheck ./scripts/*.sh
> 
> In ./scripts/createAliases.sh line 6:
> LLVMFILES=/usr/bin/llvm*
>           ^------------^ SC2[12](https://github.com/AFLplusplus/LibAFL/actions/runs/10873386066/job/30169439681?pr=2522#step:7:13)5 (warning): Brace expansions and globs are literal in assignments. Quote it or use an array.
> 
> 
> In ./scripts/createAliases.sh line 7:
> CLANGFILES=/usr/bin/clang*
>            ^-------------^ SC2125 (warning): Brace expansions and globs are literal in assignments. Quote it or use an array.
> 
> 
> In ./scripts/createAliases.sh line [15](https://github.com/AFLplusplus/LibAFL/actions/runs/10873386066/job/30169439681?pr=2522#step:7:16):
> 	echo "linking" $f "to" $link
>                        ^-- SC2086 (info): Double quote to prevent globbing and word splitting.
>                                ^---^ SC2086 (info): Double quote to prevent globbing and word splitting.
> 
> Did you mean: 
> 	echo "linking" "$f" "to" "$link"
> 
> 
> In ./scripts/createAliases.sh line [16](https://github.com/AFLplusplus/LibAFL/actions/runs/10873386066/job/30169439681?pr=2522#step:7:17):
> 	ln -s $f $link
>               ^-- SC2086 (info): Double quote to prevent globbing and word splitting.
>                  ^---^ SC2086 (info): Double quote to prevent globbing and word splitting.
> 
> Did you mean: 
> 	ln -s "$f" "$link"
> 
> 
> In ./scripts/createAliases.sh line [17](https://github.com/AFLplusplus/LibAFL/actions/runs/10873386066/job/30169439681?pr=2522#step:7:18):
> 	if [ -e $f ]
>                 ^-- SC2086 (info): Double quote to prevent globbing and word splitting.
> 
> Did you mean: 
> 	if [ -e "$f" ]
> 
> 
> In ./scripts/createAliases.sh line [18](https://github.com/AFLplusplus/LibAFL/actions/runs/10873386066/job/30169439681?pr=2522#step:7:19):
> 	  then cp $link /usr/local/bin/
>                   ^---^ SC2086 (info): Double quote to prevent globbing and word splitting.
> 
> Did you mean: 
> 	  then cp "$link" /usr/local/bin/
> 
> For more information:
>   https://www.shellcheck.net/wiki/SC2125 -- Brace expansions and globs are li...
>   https://www.shellcheck.net/wiki/SC[20](https://github.com/AFLplusplus/LibAFL/actions/runs/10873386066/job/30169439681?pr=2522#step:7:21)86 -- Double quote to prevent globbing ...
> Error: Process completed with exit code 1.
> ```
Should be fixed now. What are your thoughts on using cargo-binstall to speed up the container building process? @tokatoka @domenukk

**Comment by tokatoka** [comment]
what does it do?

**Comment by Nereuxofficial** [comment]
> what does it do?

It would allow for the downloading of release builds of cargo subcommands from Github. In the case of sccache this would reduce the time on my local machine from over 2 Minutes to less than ten seconds.

**Comment by tokatoka** [comment]
you can add

**Comment by domenukk** [comment]
Thanks!

**Pull request merged**